Add script to do all the (first time) database initialisation

pmeulen opened the pull request:
The script can be safely run multiple times, e.g. to upgrade the database schema for new versions, or to update the middleware configuration

quartje reviewed (changes requested):
nice, this will make the initial setup easier. We also need to update the README.md to reflect this?

pmeulen commented:
> nice, this will make the initial setup easier. We also need to update the README.md to reflect this?

Yes, I had held back this change because #3 modified the README as well. I've now rebased and updated the README with the new instructions.

pmeulen commented:
As discussed in https://github.com/OpenConext/Stepup-Middleware/pull/410, running the console command as root is troublesome because it messes op the ownership of the Symfony cache. I've added chowns the the init script to fix that.

Another option would be to use sudo, but that would require adding the sudo package to the base container(s). That could be a better solution to cache ownership issue.

quartje reviewed (approved):
Looks fine to me, let's merge! As discussed, I will add a story to the backlog to make an entrypoint on the Symfony dockers that creates and chowns cache during startup of the container

MKodde commented on stepup/README.md:
````diff
@@ -1,27 +1,27 @@
 ```
 
-Secondly you need to create the `stepup/gateway/surfnet_yubikey.yaml` filewith your Yubikey API credentials. If you do not have API credentials, you can get them at <https://upgrade.yubico.com/getapikey/>. You require a Yubikey to get an API key.
+Secondly you need to create the `stepup/gateway/surfnet_yubikey.yaml` filew ith your Yubikey API credentials. If you do not have API credentials, you can get them at <https://upgrade.yubico.com/getapikey/>. You require a Yubikey to get an API key.
````
Typo: ~filew ith~ file with

MKodde commented on stepup/init-db.sh:
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -e
```
Curiosity: why do you instruct the script to immediately exit upon error in this script, and not in the others? Is there a distinct advantage for doing it here?

TIL:  `set -x` is going to help me in the future while writing `bash`

MKodde commented on stepup/README.md:
````diff
@@ -30,36 +30,34 @@ surfnet_yubikey_api_client:
+This associates the admin account with your Yubikey. The account has SRAA rights. You can then use your Yubikey to log in as user `admin` with password `admin`. Note that you must run this command before the first time you log in with the admin account to the SA or RA, otherwise this command will fail because the admin identity already exists.
+
+```shell
+docker compose exec middleware /var/www/html/bin/console middleware:bootstrap:identity-with-yubikey urn:collab:person:dev.openconext.local:admin dev.openconext.local "Your Name" Your@email en_GB Yubikey_ID
````
Did you intentionally not move this to a script? I'm totally fine with having it in the README as it also ensures you need to remain engaged in order to not mis a step.

MKodde commented on stepup/README.md:
```diff
@@ -30,36 +30,34 @@ surfnet_yubikey_api_client:
-Mailcatcher is included. You can view the email by going to http://localhost:1080
+You can now login to the self-service portal at <https://selfservice.dev.openconext.local> and login to the RA portal at <https://ra.dev.openconext.local> using your admin/admin account.
+
+Mailcatcher is included. You can view the email by going to <http://localhost:1080>
```
Not for this PR: @quartje can we give her a hostname?

MKodde reviewed (approved):
Nice optimizations!
